modules/generator/generator_test.go

```diff
@@ -1,217 +0,0 @@
-package generator
-
-import (
-	"context"
-	"flag"
-	"fmt"
-	"net/http"
-	"net/http/httptest"
-	"net/url"
-	"os"
-	"testing"
-	"time"
-
-	gokitlog "github.com/go-kit/log"
-	"github.com/go-kit/log/level"
-	"github.com/grafana/dskit/flagext"
-	"github.com/grafana/dskit/kv/consul"
-	"github.com/grafana/dskit/ring"
-	"github.com/grafana/dskit/services"
-	"github.com/prometheus/client_golang/prometheus"
-	prometheus_common_config "github.com/prometheus/common/config"
-	"github.com/prometheus/common/model"
-	prometheus_config "github.com/prometheus/prometheus/config"
-	"github.com/prometheus/prometheus/storage/remote"
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
-	"github.com/weaveworks/common/user"
-
-	"github.com/grafana/tempo/modules/overrides"
-	"github.com/grafana/tempo/pkg/tempopb"
-	v1 "github.com/grafana/tempo/pkg/tempopb/trace/v1"
-	"github.com/grafana/tempo/pkg/util"
-	"github.com/grafana/tempo/pkg/util/log"
-	"github.com/grafana/tempo/pkg/util/test"
-)
-
-const localhost = "localhost"
-
-type metric struct {
-	name string
-	val  float64
-}
-
-func TestGenerator(t *testing.T) {
```

Comment:
I've deleted these tests in favour of the e2e tests. These tests have to do a lot of config management/mocking to get the generator running and verify the metrics emitted are correctly. This is exactly the same as what the e2e tests do, but their code is a bit simpler.

If anyone feels strong about keeping them, I don't mind reinstating them.